### PR TITLE
Fix Github CI builds and modulesync

### PR DESCRIPTION
**Pull request by alexjfisher**
This PR is a continuation of https://github.com/voxpupuli/puppet-gitlab_ci_runner/pull/93 (which I accidentally got into a closed state I couldn't reopen).

### CODE REVIEW

**Comment by alexjfisher** [comment]
@trevor-vaughan Hi!  Any chance you could spare some beaker docker expertise?

Our github workflow has a 'gitlab' service started, but the beaker container doesn't seen to have any network connectivity with it.
(Also see this [thread](https://github.com/voxpupuli/puppet-gitlab_ci_runner/pull/93#discussion_r531802943))

**Comment by baurmatt** [comment]
I tried to debug this a little bit further. Some things I've learned:
* Github Actions creates a dedicated bridge network in which it runs services
* Beaker containers are started in another bridge network, making it impossible to reach the Gitlab service running in the dedicated bridge network
* Github Actions seems to have no way to change the dedicated network
* We might be able to switch the Beaker container to the Github Actions network by leveraging the `dockeropts` config option. But I don't understand the syntax.
* Using the "`voxpupuli/puppet-vault_lookup`" solution (creating a dedicated beaker nodeset) seems to be not an option because we would need to do it for each supported OS.

File: .msync.yml [hunk 1]
```diff
@@ -1 +1,2 @@
-modulesync_config_version: '3.0.0'
+---
+modulesync_config_version: '4.0.0'
```
Comment (ekohl) [inline .msync.yml:2]: Can we update to 4.1.0? :)

File: spec/spec_helper_acceptance.rb [hunk 2]
```diff
@@ -31,28 +31,25 @@
...
-  file { [ '/root/.puppetlabs', '/root/.puppetlabs/bolt']:
+  file { [ '/root/.puppetlabs', '/root/.puppetlabs/bolt', '/root/.puppetlabs/etc', '/root/.puppetlabs/etc/bolt']:
     ensure => directory,
   }
 
-  file { '/root/.puppetlabs/bolt/analytics.yaml':
+  # Needs to existing to not trigger a warning sign...
+  file { '/root/.puppetlabs/etc/bolt/analytics.yaml':
     ensure  => file,
-    content => "disabled: true\n",
```
Comment (ekohl) [inline spec/spec_helper_acceptance.rb:52]: Can we keep this part?

**Review by ekohl — COMMENTED** [review]
Thanks for looking into this! I guess I was trying to be too ambitious by using native GHA.

File: spec/spec_helper_acceptance.rb (hunk 2 above)
Comment (baurmatt) [reply spec/spec_helper_acceptance.rb:52]: We do :) It's moved to the `bolt-project.yaml` :)

Comment (ekohl) [reply spec/spec_helper_acceptance.rb:52]: Ah right ...

**Comment by ekohl** [comment]
Let's see if I broke it again with my latest change. Feel free to drop that commit if it doesn't.

File: spec/spec_helper_acceptance.rb [hunk 3]
```diff
@@ -1,66 +1,7 @@
 require 'voxpupuli/acceptance/spec_helper_acceptance'
 
+ENV['BEAKER_FACTER_GITLAB_IP'] = File.read(File.expand_path('~/GITLAB_IP')).chomp
```
Comment (ekohl) [inline spec/spec_helper_acceptance.rb:3]: https://github.com/voxpupuli/voxpupuli-acceptance/#environment-variables-to-facts

I also thought about doing it in the workflow file but I think this is better for users who want to run it outside of GHA.

File: spec/setup_acceptance_node.pp [hunk 4]
```diff
@@ -0,0 +1,48 @@
+# The omnibus installer use the following algorithm to know what to do.
```
Comment (ekohl) [inline spec/setup_acceptance_node.pp:1]: See https://github.com/voxpupuli/voxpupuli-acceptance/#per-host-configuration for how this file is used.

**Review by ekohl — APPROVED** [review]
I rebased it to modulesync 4.1.0 and also squashed some commits. I think this is now ready to merge if it's green.

**Comment by ekohl** [comment]
We shouldn't forget to create issues that we've dropped Amazon Linux and EL6 for the changelog.

**Comment by alexjfisher** [comment]
I think the next release will have a substantial release summary.  I'm hoping to get PR together based on https://github.com/syseleven/puppet-gitlab_ci_runner/tree/clean-move-to-concat-deferred  At the moment, the config file is only written when registering a runner, and any subsequent config changes you make have no effect.  There are also plenty of more exotic configurations you can't make whilst registering.

**Comment by alexjfisher** [comment]
For completeness (and mainly because I pinged Trevor and don't want to waste his time!), in https://github.com/voxpupuli/puppet-gitlab_ci_runner/pull/105 I managed to get the docker networking working in the way we originally wanted.  Might revisit this at a later date.

**Pull request merged**